relay-server/src/actors/processor.rs

```diff
@@ -2351,9 +2351,20 @@ impl EnvelopeProcessorService {
             SamplingResult::Drop(rule_ids)
                 if state.event_type() == Some(EventType::Transaction) =>
             {
-                state
-                    .managed_envelope
-                    .reject(Outcome::FilteredSampling(rule_ids.clone()));
+                let feature_enabled = false;
+                let outcome = Outcome::FilteredSampling(rule_ids.clone());
+                state.managed_envelope.retain_items(|item| {
+                    if feature_enabled && item.ty() == &ItemType::Profile {
+                        item.set_header("sampled", false);
```

Comment:
@viglia you can access this header through `item.get_header()` in https://github.com/getsentry/relay/blob/a4a6922815903f14528ffb5ca7437573c9192ee5/relay-server/src/actors/store.rs#L617 and add the flag to the kafka message.